src/test/java/com/agnieszkapawska/flashcards/FlashcardsApplicationAbstractTests.java

```diff
@@ -0,0 +1,39 @@
+package com.agnieszkapawska.flashcards;
+
+import com.agnieszkapawska.flashcards.domain.dtos.FlashcardDto;
+import com.agnieszkapawska.flashcards.domain.dtos.FlashcardSaveResponseDto;
+import com.agnieszkapawska.flashcards.domain.facades.FlashcardFacade;
+import org.junit.Before;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit4.SpringRunner;
+
+@ActiveProfiles(profiles = "test")
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT, classes = FlashcardsApplication.class)
+@RunWith(SpringRunner.class)
+@TestPropertySource(locations = "classpath:application-tests.properties")
+public abstract class FlashcardsApplicationAbstractTests {
+	@LocalServerPort
+	private int serverPort;
+	protected String baseUrl;
+	protected FlashcardDto flashcardDto;
+	protected FlashcardSaveResponseDto flashcardSaveResponseDto;
+	@MockBean
+	protected FlashcardFacade flashcardFacade;
```

Comment:
Maybe let's move it back to `FlashcardControllerTest`. It won't be needed in all test classes.